src/MLJMultivariateStatsInterface.jl

```diff
@@ -150,4 +150,1052 @@ metadata_pkg.(
+non-Gaussian and independent from each other.
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+    mach = machine(model, X)
+
+Where
+
+- `X`: is any table of input features (eg, a `DataFrame`) whose columns
+  are of scitype `Continuous`; check the scitype with `schema(X)`
+
+# Hyper-parameters
+
+- `k::Int=0`: The number of independent components to recover, set automatically if `0`.
+- `alg::Symbol=:fastica`: The algorithm to use (only `:fastica` is supported at the moment).
+- `fun::Symbol=:tanh`: The approximate neg-entropy function, one of `:tanh`, `:gaus`.
+- `do_whiten::Bool=true`: Whether or not to perform pre-whitening.
+- `maxiter::Int=100`: The maximum number of iterations.
+- `tol::Real=1e-6`: The convergence tolerance for change in matrix W.
+- `mean::Union{Nothing, Real, Vector{Float64}}=nothing`: mean to use, if nothing (default)
+   centering is computed and applied, if zero, no centering, a vector of means can
+   be passed.
+- `winit::Union{Nothing,Matrix{<:Real}}=nothing`: Initial guess for matrix `W` either
+   an empty matrix (random initilization of `W`), a matrix of size `k × k` (if `do_whiten`
+   is true), a matrix of size `m × k` otherwise. If unspecified i.e `nothing` an empty
+   `Matrix{<:Real}` is used.
+
+# Operations
+
+- `transform(mach, Xnew)`: Return lower dimensional projection of the target given new
+  features `Xnew` having the same Scitype as `X` above.
+
+# Fitted parameters
+
+The fields of `fitted_params(mach)` are:
+
+ BUG: Does not have a projection class. It would also be cool to see the whitened
+matrix in fitted_params, to show how the covariance is the identity
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `indim`: Dimensions of the provided data.
+- `outdim`: Dimensions of the transformed result.
+- `mean`: The mean vector.
+
+# Examples
+
+```
+using MLJ
+using LinearAlgebra
+
+ICA = @load ICA pkg=MultivariateStats
+
+X, y = @load_iris
+
+model = ICA(k = 2, tol=0.1)
+mach = machine(model, X) |> fit!
+
+projection = transform(mach, X)
+```
+
+See also
+[`PCA`](@ref), [`KernelPCA`](@ref), [`FactorAnalysis`](@ref), [`PPCA`](@ref)
+"""
+ICA
+"""
+$(MMI.doc_header(LDA))
+
+`LDA`: Multiclass linear discriminant analysis. The algorithm learns a
+projection matrix `P` that projects a feature matrix `Xtrain` onto a lower dimensional
+space of dimension `out_dim` such that the trace of the transformed between-class
+scatter matrix(`Pᵀ*Sb*P`) is maximized relative to the trace of the transformed
+within-class scatter matrix (`Pᵀ*Sw*P`).The projection matrix is scaled such that
+`Pᵀ*Sw*P=I` or `Pᵀ*Σw*P=I`(where `Σw` is the within-class covariance matrix) .
+Predicted class posterior probability for feature matrix `Xtest` are derived by
+applying a softmax transformationto a matrix `Pr`, such that  rowᵢ of `Pr` contains
+computed distances(based on a distance metric) in the transformed space of rowᵢ in
+`Xtest` to the centroid of each class.
+
```

Comment:
I feel this introductory section is either too short or too long. 

If it aims to be self complete, it fails because it defines the algorithm for finding the projection matrix in terms of undefined objects such as the scattering matrices. But it's already quite long, and I'm guessing this algorithm is pretty standard. 

The second part about classification is worth keeping, as this is not implemented in the core algorithm but added in the MLJ interface.  But I find the docstring a little unclear. 

There is also the possibility of confusion because `Xtrain`/`Xtest` would be tables in practice, but are matrices in this doc-string. 

So my suggestion is to collapse the first part and re-work the second. Something along these lines:

---

LDA multiclass linear descriminant analysis learns a projection in a space of features to a lower dimensional space, in a way that attempts to preserve as much as possible the degree to which the target classes are separable [ADD WIKI OR PAPER REFERENCE]. This can be used either for dimension reduction of the features (see `transform` below) or for probabilistic classification of the target (see `predict` below). 

In the case of prediction, the class probability for a new observation reflects the proximity of that observation to training observations associated with that class, and how far away the observation is from those associated with other classes. Specifically, the distances, in the transformed (projected) space, of a new observation, from the centroid of each target class, is computed; the resulting vector of distances (times minus one) is passed to a softmax function to obtain a class probability prediction. Here "distance" is computed using a user-specified distance function. 
---